src/main.c

```diff
@@ -112,76 +104,29 @@ void main(void)
  */
 void handle_timer(struct k_timer *dummy)
 {
-	int pin_in_val = gpio_pin_get_raw(gpio0, SENS_PIN);
+	char *err = NULL;
 
-	double flow_rate = atomic_get(&interrupt_count) / TIKSCONF;
-	char buf[NMEA_LENGTH] = "";
+	// TODO: Should check err if atomic_get can fail.
+	const uint32_t ticks = d39_gpio_getticks(&err);
+	const double flow_rate = ticks / TIKSCONF;
 
+	char buf[NMEA_LENGTH] = "";
 	nmeaflow_buildmsg(buf, flow_rate);
 
-	printk("Update - %d (%d), %s\n", pin_in_val, atomic_get(&interrupt_count), buf);
-
-	atomic_set(&interrupt_count, (atomic_val_t)0);
+	printk("Update - ints: %d, %s\n", ticks, buf);
 
 	if (notify_enable) {
 		bt_gatt_notify(NULL, &d39fuelsensor_svc.attrs[1], &buf, 15);
 	}
-
-}
-
-/*
- * setup gpio0 SENS_PIN
- */
-const struct device *initialize_gpio()
-{
-
-	int ret;
-
-	gpio0 = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
-	if (gpio0 == NULL) {
-		printk("Could not get %s device\n", DT_LABEL(DT_NODELABEL(gpio0)));
-		return NULL;
-	}
-
-	ret = gpio_pin_configure(gpio0, SENS_PIN, GPIO_INPUT | GPIO_PULL_UP);
-	if (ret) {
-		printk("Failed to configure GPIO input (%d).", ret);
-		return NULL;
-	}
-
-	if (gpio_pin_interrupt_configure(gpio0, SENS_PIN, GPIO_INT_TRIG_LOW)) {
-		printk("Failed to configure GPIO interrupt.\n");
-		return NULL;
-	}
-
-	gpio_init_callback(&tick_cb_data, handle_tick, BIT(SENS_PIN));
-	gpio_add_callback(gpio0, &tick_cb_data);
-
-	return gpio0;
-}
-
-/*
- * increase counter on each interrupt for SENS_PIN
- */
-void handle_tick(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
-{
-	atomic_inc(&interrupt_count);
 }
 
 /*
  * initialize bluetooth services
  */
-void bt_ready(int err)
+void bt_ready()
```

Comment:
this is in principle a callback function, that should be given to the bluetooth initialization, therefore it has the `int err` as parameter.

Reply:
and it should be probably moved to bt files as well